### PR TITLE
Bitnodes match against altnames so anti-spam and other hosting works.  Fix short read error.

### PR DESCRIPTION
**Pull request by gandrewstone**
Fix the short read error (fix provided by @bitcartel).

### CODE REVIEW

File: src/bitnodes.cpp [hunk 1]
```diff
@@ -172,7 +199,7 @@ class client
             } else {
                 timer_.get()->cancel();
                 if (error != boost::asio::error::eof) {
-                    throw runtime_error(strprintf("Bitnodes HTTP read error: %s\n", error.message().c_str()));
+                    throw runtime_error(strprintf("Bitnodes HTTP read error: %s. Bytes received: %d\n", error.message().c_str(),bytes_transferred));
```
Comment (bitcartel) [inline src/bitnodes.cpp]: space after , --> c_str(),bytes_

File: src/bitnodes.cpp [hunk 2]
```diff
@@ -53,23 +53,46 @@ class client
...
-            timer_.get()->cancel();
+            return true;
         }
 
-        void run(boost::asio::io_service& io_service)
-        {
-            timer_.get()->async_wait(boost::bind(&client::timeout_handler, this, boost::asio::placeholders::error));
-            io_service.run();
-            timer_.get()->cancel();
+            ~client() { timer_.get()->cancel(); }
```
Comment (bitcartel) [inline src/bitnodes.cpp]: This diff chunk with ~client() and run() seems unnecessary - it's just formatting.

**Review by bitcartel — APPROVED** [review]
ACK, see comments about formatting.

**Review by ptschip — COMMENTED** [review]
As for the formatting i think we should just run these files through clang-format and add them to the list of formatted files.  ( I think this one never got done)

**Comment by gandrewstone** [comment]
@bitcartel @ptschip ok, I've added these 2 files to the clang-formatter.

Its unfortunately still converting to a single-line destructor, which would not be my choice but I don't think it matters that much...

I'm going to merge once the regression tests pass since this latest commit is syntax-only

**Review by ptschip — APPROVED** [review]

**Pull request merged**